lib/config/validation.ts

```diff
@@ -672,3 +655,45 @@ export async function validateConfig(
   warnings.sort(sortAll);
   return { errors, warnings };
 }
+
+function validateRegexManagerFields(
+  regexManager: RegexManagerConfig,
+  currentPath: string,
+  errors: ValidationMessage[]
+): void {
+  if (is.nonEmptyArray(regexManager.matchStrings)) {
+    for (const matchString of regexManager.matchStrings) {
+      try {
+        regEx(matchString);
+      } catch (e) {
+        errors.push({
+          topic: 'Configuration Error',
+          message: `Invalid regExp for ${currentPath}: \`${String(
+            matchString
+          )}\``,
+        });
+      }
+    }
+  } else {
+    errors.push({
+      topic: 'Configuration Error',
+      message: `Each Regex Manager must contain a non-empty matchStrings array`,
+    });
+  }
+
+  const mandatoryFields = ['depName', 'currentValue', 'datasource'];
+  for (const field of mandatoryFields) {
+    const templateField = `${field}Template` as keyof RegexManagerTemplates;
+    if (
+      !regexManager[templateField] &&
+      !regexManager.matchStrings.some((matchString: string) =>
```

Comment:
```suggestion
      !regexManager.matchStrings.some((matchString) =>
```
do we really need the type here?